### ISSUE TITLE
Bug with and anchor_x="center"

### ISSUE THREAD

**Issue by gran4** [pyglet/pyglet#915]
**Describe the bug**
The Text is not in the center as you would expect it. That is because it is calculating the center wrong using the width.
This assumes the person put in the exact width of the text, which is often not the case

**System Information:**
Mac M1 2020. Another person was having the issue as well(not sure about his system)

**How To Reproduce**
A small code example that illustrates the bug is very useful to help triage.

I HAVE CONFIRMED IT IS A PYGLET ISSUE. LOOK AT:
https://github.com/pythonarcade/arcade/issues/1839

Line 1411 `return self._x - width // 2`
seems to be the issue

Arcade dev.25 seems to have updated pyglet

**CODE**:
```
import arcade


class IssueDemonstrationWindow(arcade.Window):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.text = arcade.Text(
            text="MIDDLE",
            start_x=400,
            start_y=300,
            font_size=72,
            width=8000,
            anchor_x="center",
            align="center",
        )
        self.set_viewport(
            left=0,
            right=800,
            bottom=0,
            top=600,
        )

    def on_draw(self):
        self.text.draw_debug()


window = IssueDemonstrationWindow()
arcade.run()
```

**Comment by caffeinepills** [comment]
This is expected behavior.

In your example, the width is 8000, and your alignment is center.

Either remove the width, or remove the alignment.

**Comment by gran4** [comment]
> This is expected behavior.
> 
> In your example, the width is 8000, and your alignment is center, which will cause it to go far off the window.

I did that on purpose, just to show that the width is what controls it on anchor_x="center".
Also, I looked at the code, and it looked like it is a pyglet thing because the arcade class is only a light wrapper.

**Comment by caffeinepills** [comment]
> I did that on purpose, just to show that the width is what controls it on anchor_x="center".
I see, that makes sense.

This is normal. Per the docs for `anchor_x`:
https://pyglet.readthedocs.io/en/latest/modules/text/layout.html#pyglet.text.layout.TextLayout.anchor_x

> For the purposes of calculating the position resulting from this alignment, the width of the layout is taken to be width if multiline is True and wrap_lines is True, otherwise content_width.

In other words, `width` is used when multiline is True, otherwise it bases the positioning off of `content_width`. If you add `multiline=True` to your example, it should behave how you are wanting.

**Comment by gran4** [comment]
So it is intended?

**Comment by bunny-therapist** [comment]
This appears to have changed sometime between pyglet 2.0.9 and 2.0.dev23. Could you confirm this was also intentional?

**Comment by caffeinepills** [comment]
Yes, this is behavior is normal and intended.

**Comment by gran4** [comment]
Thank you

**Comment by pushfoo** [comment]
tl;dr 
1. Are we using the current approach because it seems to have a clear way of automatically breaking lines?
2. Should we consider other centering approaches?

To my understanding, there are two ways to center things:
1. Relative to a width (as we're doing now)
2. For each line:
    1. Calculate the line's width
    2. Divide it by two
    3. Shift the line's start point relative to the middle position

The downside of the second option is that it doesn't really allow for handling line breaks if implemented naively.
